### PR TITLE
Update documentation - intro and folding sections

### PR DESCRIPTION
**Pull request by sdondley**
See #692

### CODE REVIEW

File: doc/vimwiki.txt [hunk 1]
```diff
@@ -1471,11 +1471,18 @@ Tags-related commands and options:
...
+control how folds behave with by setting the |g:vimwiki_folding| variable to
+the desired value in your configuration file:
 
-Example for list folding with |g:vimwiki_folding| set to 'list':
+let g:vimwiki_folding = <value>    " Surround value with quotes.
+
+If left unset, folding will default to Vim's "manual" method. If you wish to
+turn folding off completely, set |g:vimwiki_folding| to `''`(an empty string).
+
+Here is example of how folds work with |g:vimwiki_folding| set to 'list':
```
Comment (hq6) [inline doc/vimwiki.txt]: "Here is *an* example"

**Comment by sdondley** [comment]
One more fix. Sorry!

**Comment by hq6** [comment]
Here's how you can clean up the commit history after you are happy with the state of the change.
```
git rebase -i <sha_of_commit_before_your_first_commit>
# Change the first word on all the lines except the first to `squash`.
# Save the file
```

**Comment by sdondley** [comment]
OK, thanks. I definitely need to advance my git skills.

**Comment by ranebrown** [comment]
@hq6 it might be easier to just do a squash and merge (I didn't know about this until recently). Up to you though.

**Comment by sdondley** [comment]
Yeah, I'll take some time this weekend to wrap my head around this. Thanks for your patience.

**Comment by hq6** [comment]
@ranebrown I'm down to try it in the future (I typically merge PR's manually on the command line since it feels like there's less magic), but for now let's give Steve an opportunity to learn how to squash, since I think it will be beneficial for him (and us) in the future.

**Comment by sdondley** [comment]
Looks like I'll be using vimwiki a lot. To get more familiar with it, I'm going to go through 1 or 2 help sections per day and look to see if there is room for improvement. I'll start a new branch called `doc_updates` for this purpose.

Let me know if there is any area of the docs that needs special attention/care.

**Comment by ranebrown** [comment]
@sdondley Are you still planning on cleaning up the commit history? (Conflicts need resolved as well)

**Comment by sdondley** [comment]
Sorry, I thought the squashing was something you had to do on the official vimwiki repo. Am I wrong?

**Comment by ranebrown** [comment]
See [this comment](https://github.com/vimwiki/vimwiki/pull/694#issuecomment-488794357). We can sqash and merge but you can also edit your commit history using rebase which means it is less work on our end.

**Comment by sdondley** [comment]
OK, will work on it this weekend.

**Pull request closed without merging**